### PR TITLE
chore: visual refresh on radio component

### PR DESCRIPTION
**Pull request by khamudom**
# Pull Request

## 📖 Description

Updated the size to 18px.

before unchecked
image

after unchecked
image

before checked
image

after checked
image

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-components/src/radio-group/radio-group.styles.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export const radioGroupStyles: (
         align-items: flex-start;
         margin: calc(${designUnit} * 1px) 0;
```
Comment (chrisdholt) [inline packages/web-components/fast-components/src/radio-group/radio-group.styles.ts:22]: Created https://github.com/microsoft/fast/issues/5407 to address this.

File: packages/web-components/fast-components/src/radio/radio.styles.ts [hunk 2]
```diff
@@ -42,174 +38,155 @@ export const radioStyles: (
...
+    :host(:enabled:hover) .control {
+        border-color: ${strokeControlStrongHover};
+    }
+    :host(:enabled:active) .control {
+        background: ${neutralFillInputActive};
+        border-color: ${strokeControlStrongActive};
+    }
+    :host(:${focusVisible}) .control {
+        border-color: ${strokeControlStrongFocus};
+        outline: calc(${focusStrokeWidth} * 1px) solid ${strokeControlStrongFocus};
```
Comment (bheston) [inline packages/web-components/fast-components/src/radio/radio.styles.ts:76]: This should be the focus stroke color. Will likely both look the same, but all focus treatments should use the same focus-specific recipes.

**Review by bheston — APPROVED** [review]

File: packages/web-components/fast-components/src/radio/radio.styles.ts (hunk 2 above)
Comment (khamudom) [reply packages/web-components/fast-components/src/radio/radio.styles.ts]: updated

File: packages/web-components/fast-components/src/radio/radio.styles.ts [hunk 3]
```diff
@@ -42,174 +42,155 @@ export const radioStyles: (
...
         outline: none;
         margin: calc(${designUnit} * 1px) 0;
-        /* Chromium likes to select label text or the default slot when
-            the radio is clicked. Maybe there is a better solution here? */
+        ${
+            /*
+             * Chromium likes to select label text or the default slot when
+             * the radio button is clicked. Maybe there is a better solution here?
+             */ ""
+        }
```
[the remark this answers is not in the document]
Comment (khamudom) [reply packages/web-components/fast-components/src/radio/radio.styles.ts]: updated

File: packages/web-components/fast-components/src/radio/radio.styles.ts [hunk 4]
```diff
@@ -41,175 +41,178 @@ export const radioStyles: (
...
+            cursor: ${disabledCursor};
+        }
+
+        :host(.checked) slot[name="checked-indicator"] {
+            opacity: 1;
+        }
+
+        :host(.disabled) {
+            opacity: ${disabledOpacity};
+        }
```
Comment (radium-v) [inline packages/web-components/fast-components/src/radio/radio.styles.ts:154]:
```suggestion
        :host([disabled]) .label,
        :host([readonly]) .label,
        :host([readonly]) .control,
        :host([disabled]) .control {
            cursor: ${disabledCursor};
        }

        :host(.checked) slot[name="checked-indicator"] {
            opacity: 1;
        }

        :host([disabled]) {
            opacity: ${disabledOpacity};
        }
```

File: packages/web-components/fast-components/src/radio/radio.styles.ts [hunk 5]
```diff
@@ -41,175 +41,178 @@ export const radioStyles: (
...
+                :host(.checked.disabled:hover) .control,
+                .control:active {
+                    background: ${SystemColors.Field};
+                    border-color: ${SystemColors.GrayText};
+                }
+
+                :host(.disabled) slot[name="checked-indicator"],
+                :host(.checked.disabled) .control:hover slot[name="checked-indicator"] {
+                    fill: ${SystemColors.GrayText};
+                }
```
Comment (radium-v) [inline packages/web-components/fast-components/src/radio/radio.styles.ts:215]:
```suggestion
                :host([disabled]) {
                    opacity: 1;
                }

                :host([disabled]) .label {
                    color: ${SystemColors.GrayText};
                }

                :host([disabled]) .control,
                :host(.checked[disabled]:hover) .control,
                .control:active {
                    background: ${SystemColors.Field};
                    border-color: ${SystemColors.GrayText};
                }

                :host([disabled]) slot[name="checked-indicator"],
                :host(.checked[disabled]) .control:hover slot[name="checked-indicator"] {
                    fill: ${SystemColors.GrayText};
                }
```

File: packages/web-components/fast-components/src/radio/radio.styles.ts [hunk 6]
```diff
@@ -41,175 +41,178 @@ export const radioStyles: (
...
+            cursor: pointer;
+            font-size: ${typeRampBaseFontSize};
+            line-height: ${typeRampBaseLineHeight};
+        }
+
+        slot[name="checked-indicator"] {
+            flex-shrink: 0;
+        }
+
+        slot[name="checked-indicator"] {
```
Comment (radium-v) [inline packages/web-components/fast-components/src/radio/radio.styles.ts:126]: These selectors can be combined.

**Review by radium-v — CHANGES_REQUESTED** [review]
I'm not sure which of `.checked`, `[aria-checked="true"]`, and `[checked]` are the preferred way to style the checked state.

File: packages/web-components/fast-components/src/radio/radio.styles.ts (hunk 4 above)
Comment (khamudom) [reply packages/web-components/fast-components/src/radio/radio.styles.ts]: updated

File: packages/web-components/fast-components/src/radio/radio.styles.ts (hunk 5 above)
Comment (khamudom) [reply packages/web-components/fast-components/src/radio/radio.styles.ts]: updated

File: packages/web-components/fast-components/src/radio/radio.styles.ts (hunk 6 above)
Comment (khamudom) [reply packages/web-components/fast-components/src/radio/radio.styles.ts]: combined

**Comment by EisenbergEffect** [comment]
> I'm not sure which of `.checked`, `[aria-checked="true"]`, and `[checked]` are the preferred way to style the checked state.

I think in general we wanted to move away from putting classes on the host or driving anything based on those, in order to not get into potential conflicts with vdom systems or other frameworks that want to manage all aspects of the host element.

**Comment by chrisdholt** [comment]
> > I'm not sure which of `.checked`, `[aria-checked="true"]`, and `[checked]` are the preferred way to style the checked state.
> 
> I think in general we wanted to move away from putting classes on the host or driving anything based on those, in order to not get into potential conflicts with vdom systems or other frameworks that want to manage all aspects of the host element.

I'd personally suggest binding to the `aria-checked="true"` for now. It mirrors how we've approached other instances and it maintains a direct tie to what makes this "checked". Were the API to change at some point away from "checked", the selector would remain for a11y and wouldn't necessarily require changes.

**Comment by bheston** [comment]
> > > I'm not sure which of `.checked`, `[aria-checked="true"]`, and `[checked]` are the preferred way to style the checked state.
> > 
> > 
> > I think in general we wanted to move away from putting classes on the host or driving anything based on those, in order to not get into potential conflicts with vdom systems or other frameworks that want to manage all aspects of the host element.
> 
> I'd personally suggest binding to the `aria-checked="true"` for now. It mirrors how we've approached other instances and it maintains a direct tie to what makes this "checked". Were the API to change at some point away from "checked", the selector would remain for a11y and wouldn't necessarily require changes.

Can we get this written up as a set of principles and priorities. Like use native attributes first, followed by aria, followed by part, then class, etc.? This question keeps coming up and rules are getting applied subjectively. And honestly half the time I'm not sure what's best.

**Comment by khamudom** [comment]
updated to use `[aria-checked="true"]`

**Pull request merged**